lib/src/app/app.dart

```diff
@@ -86,4 +87,25 @@ class AppClient {
     final stream = response.map((event) => event.logs);
     return stream.asBroadcastStream(onCancel: (_) => response.cancel());
   }
+
+  /// List the [Authorization]s available for the currently authenticated user
+  Future<List<Authorization>> listAuthorizations(String organizationId, {List<String> resourceIds = const []}) async {
+    final request = ListAuthorizationsRequest()
+      ..organizationId = organizationId
+      ..resourceIds.addAll(resourceIds);
+    final response = await _client.listAuthorizations(request);
+    return response.authorizations;
+  }
+
+  Future<List<Permission>> checkPermissions(ResourceType resourceType, String resourceId, List<Permission> permissions) async {
+    final request = CheckPermissionsRequest()
+      ..permissions.add((AuthorizedPermissions()
+        ..resourceType = resourceType.name
+        ..resourceId = resourceId
+        ..permissions.addAll(permissions.map((e) => e.value))));
+    final response = await _client.checkPermissions(request);
+    return response.authorizedPermissions.first.permissions
+        .map((e) => Permission.values.firstWhere((element) => element.value == e))
+        .toList();
```

Comment:
[q] (mainly because I'm unfamiliar with the shape of the API response) - why are we only considering the first authorized permission in the response? (`response.authorizedPermissions.first`)

Reply:
This is a great question! It's because the API actually takes in an array of of permissions to check, and then returns an array of validated permissions. 

This method only allows for a single permission check, so we will only get (at most) a single permission response back